compair/static/modules/answer/answer-module.js

```diff
@@ -242,61 +242,74 @@ module.controller(
             }
         }
 
-        $scope.answerSubmit = function (saveDraft) {
+        $scope.answerSubmit = function (saveDraft, answerForm) {
             $scope.submitted = true;
             $scope.saveDraft = saveDraft;
-            var wasDraft = $scope.answer.draft;
 
-            var file = answerAttachService.getFile();
-            if (file) {
-                $scope.answer.file = file;
-                $scope.answer.file_id = file.id
-            } else if ($scope.answer.file) {
-                $scope.answer.file_id = $scope.answer.file.id;
-            } else {
-                $scope.answer.file_id = null;
-            }
-
-            // copy answer so we don't lose draft state on failed submit
-            var answer = angular.copy($scope.answer);
-            answer.draft = !!saveDraft;
-            answer.tracking = $scope.tracking.toParams();
-
-            AnswerResource.save({'courseId': $scope.courseId, 'assignmentId': $scope.assignmentId}, answer).$promise.then(
-                function (ret) {
-                    $scope.answer = ret;
-                    $scope.preventExit = false; // user has saved answer, does not need warning when leaving page
-                    $scope.refreshOnExit = ret.draft; // if draft was saved, we need to refresh to show updated data when modal is closed
+            $timeout(function() {
```

Comment:
I don't really like the way this is handled in the controller.

Instead of `ng-required="validated"` just use `required` and then in the controller check if the form is valid. It would be even better if the check could be handled by a directive placed on the form but if that's not possible having a separate generic function to check if the form is valid and handling scroll would be better than bundling it with the normal submit code (the current way will involve a lot of repeated code).

The challenging thing with my suggestion is controlling validation styling. In General we would be able to handle this by using `ng-class="{'has-error': form.$submitted && formControl.$invalid}"` on form fields (this may be a problem with things like chosen selects but you can explore that more later)